core/dataset.cpp

```diff
@@ -801,24 +851,12 @@ Dataset operator+(const Dataset &lhs, const DataConstProxy &rhs) {
   return res;
 }
 
-Dataset operator+(Dataset &&lhs, const Dataset &rhs) {
-  return apply(plus_equals, lhs, rhs);
-}
-
 Dataset operator+(Dataset &&lhs, const DatasetConstProxy &rhs) {
```

Comment:
Are you sure you want to keep the rvalue overloads? I think it would be more complex than done here: Before moving you will need to ensure that `rhs` is not referencing `lhs`.

I think I would just remove all `&&` overloads to avoid any headaches at this point.

Reply:
For now it is probably easier to do without them.